test/index.js

```diff
@@ -82,6 +82,65 @@ describe('Unifile class', function() {
 		});
 	});
 
+	describe('ext()', function() {
+		let unifile;
+		let inMemoryFile = '';
+		beforeEach('Instanciation', function() {
+			unifile = new Unifile();
+			unifile.use({
+				name: 'memory',
+				getInfos: () => {return {isLoggedIn: true};},
+				readFile: (session, path) => Promise.resolve(inMemoryFile),
+				writeFile: (session, path, content) => {
+					inMemoryFile = content;
+					return Promise.resolve();
+				}
+			});
+		});
+
+		it('registers an extension', function() {
+			unifile.ext({
+				onWrite: console.log,
+				onRead: console.log
+			});
+			expect(unifile.plugins.onRead).to.have.lengthOf(1);
+			expect(unifile.plugins.onWrite).to.have.lengthOf(1);
+			expect(unifile.plugins.onRead[0]).to.equal(console.log);
+			expect(unifile.plugins.onWrite[0]).to.equal(console.log);
+		});
+
+		it('can modify the input on write action', function() {
+			unifile.ext({
+				onWrite: (input) => input.replace('a', 'b')
+			});
+			return unifile.writeFile({}, 'memory', '', 'ab')
+			.then(() => expect(inMemoryFile).to.equal('bb'));
+		});
+
+		it('can modify the input on read action', function() {
+			inMemoryFile = 'ab';
+			unifile.ext({
+				onRead: (input) => input.replace('b', 'a')
+			});
+			return unifile.readFile({}, 'memory', '')
+			.should.eventually.equal('aa');
+		});
+
+		it('follow a chain of action in order', function() {
```

Comment:
IMHO it shouldn't be possible to add several codecs, as the spec is very confusing:
For instance, if you use 2 extension, say cipher / decipher and then zip, on write it's all fine (cipher then zip), but on read, it would certainly break (decipher then unzip => boom)
I think you should either clarify the spec (and likely fix your implementation :S) or allow only 1 ext / hook.
I think allowing only 1 ext is better anyway, since you can still compose your onRead / onWrite hooks like `onWrite: compose( cipher , zip ), onRead: compose( unzip, decipher )`